core/modules/replica/HttpRequestQuery.cc

```diff
@@ -145,4 +145,15 @@ bool HttpRequestQuery::optionalBool(string const& param,
     return not (itr->second.empty() or (itr->second == "0"));
 }
 
+
+double HttpRequestQuery::requiredDouble(string const& param) const {
+    auto&& itr = _query.find(param);
+    if (itr == _query.end()) {
+        throw invalid_argument(
+                string(__func__) + " parameter '" + param + "' is missing");
+    }
+    return stod(itr->second);
```

Comment:
you're sure this is never going to throw? Or is it handled upstream?

Reply:
This function is expected to throw `std::invalid_argument` for miss-formed (or out-of-range) inputs. Exceptions of this kind are caught and properly reported (via the LSST Logger) and sent back to REST API clients as `json` object by the module execution driver:
```
// File: core/modules/replica/HttpModule.cc
void HttpModule::execute(qhttp::Request::Ptr const& req,
                         qhttp::Response::Ptr const& resp,
                         string const& subModuleName) {
    try {
        executeImpl(req, resp, subModuleName);
    } catch (invalid_argument const& ex) {
        sendError(resp, __func__, "invalid parameters of the request, ex: " + string(ex.what()));
    } catch (exception const& ex) {
        sendError(resp, __func__, "operation failed due to: " + string(ex.what()));
    }
}
```

Reply:
This is the mini-REST-API framework which I implemented to reduce boilerplate code when processing the REST requests by the Master Replication Controller.